app/grandchallenge/notifications/models.py

```diff
@@ -37,3 +130,282 @@ def _assign_permissions(self):
         assign_perm("view_notification", self.user, self)
         assign_perm("delete_notification", self.user, self)
         assign_perm("change_notification", self.user, self)
+
+    @staticmethod
+    def send(type, **kwargs):  # noqa: C901
+        verb = kwargs.pop("verb")
+        try:
+            actor = kwargs.pop("actor")
+        except KeyError:
+            actor = None
+        try:
+            action_object = kwargs.pop("action_object")
+        except KeyError:
+            action_object = None
+        try:
+            target = kwargs.pop("target")
+        except KeyError:
+            target = None
+        try:
+            description = kwargs.pop("description")
+        except KeyError:
+            description = None
+        try:
+            context_class = kwargs.pop("context_class")
+        except KeyError:
+            context_class = None
+
+        if (
+            type == NotificationType.NotificationTypeChoices.FORUM_POST
+            or type
+            == NotificationType.NotificationTypeChoices.FORUM_POST_REPLY
+            or type == NotificationType.NotificationTypeChoices.ACCESS_REQUEST
+            or type == NotificationType.NotificationTypeChoices.REQUEST_UPDATE
+            or type == NotificationType.NotificationTypeChoices.MISSING_METHOD
+        ):
+            if actor:
+                receivers = [
+                    follower
+                    for follower in followers(target)
+                    if follower != actor
+                ]
+            else:
+                receivers = followers(target)
+        elif type == NotificationType.NotificationTypeChoices.NEW_ADMIN:
+            receivers = [action_object]
+        elif (
+            type == NotificationType.NotificationTypeChoices.EVALUATION_STATUS
+        ):
+            receivers = followers(target)
+            receivers.append(actor)
+        elif type == NotificationType.NotificationTypeChoices.JOB_STATUS:
+            receivers = followers(target, flag="job-active")
+        elif (
+            type
+            == NotificationType.NotificationTypeChoices.IMAGE_IMPORT_STATUS
+        ):
+            receivers = followers(action_object)
+
+        for receiver in receivers:
+            Notification.objects.create(
+                user=receiver,
+                type=type,
+                verb=verb,
+                actor=actor,
+                action_object=action_object,
+                target=target,
+                description=description,
+                context_class=context_class,
+            )
+
+    def print_notification(self, user):  # noqa: C901
+        if self.type == NotificationType.NotificationTypeChoices.FORUM_POST:
+            return format_html(
+                "{} {} {} in {} {}.",
+                user_profile_link(self.actor),
+                self.verb,
+                mark_safe(
```

Comment:
You shouldn't need `mark_safe` in here. Add any safe markup to the string in the first argument to `format_html`.

Reply:
Actually, I think mark_safe is called by format_html by default, so I don't need to add mark_safe anywhere, I guess, right? But I can just replace all the mark_safe instances with format_html then.